src/Admin/AdminHelper.php

```diff
@@ -25,6 +25,8 @@
 use Symfony\Component\Form\FormView;
 
 /**
+ * @final since sonata-project/admin-bundle 4.0
```

Comment:
This looks weird… is it a standard practice in Symfony, maybe?

Reply:
But if it results in deprecations, that's probably quite good 👍 Is the plan to replace this with keywords on master?

Reply:
Yes, the deprecation is shown with messages like this:
```
The "Sonata\AdminBundle\Admin\AdminHelper" class is considered final since sonata-project/admin-bundle 4.0. It may change without further notice as of its next major version. You should not extend it from "App\MyCustomAdminHelper".
```
The plan is to declare these classes as final on `master`.

Reply:
Just for the records, the deprecation message is built by [`DebugClassLoader`](https://github.com/symfony/symfony/blob/288bc24d0362fe87c4eb5c67f89d08f9b93e5807/src/Symfony/Component/Debug/DebugClassLoader.php#L270).

Reply:
Ok, I think you should use 3.x instead of 4.0 here, WDYT?

Reply:
Yes, the message could be confusing otherwise. Should I add the "NEXT_MAJOR" comments too or this should be enough?

Reply:
I think we can simply add `@final` to our list of things to grep for before releasing a new major. That list should be in the CONTRIBUTING.md